bdk-jvm/lib/src/test/kotlin/org/bitcoindevkit/OfflinePersistenceTest.kt

```diff
@@ -40,4 +40,31 @@ class OfflinePersistenceTest {
             actual = addressInfo.address.toString(),
         )
     }
+
+    @Test
+    fun testPersistenceWithDescriptor() {
+        val connection = Connection(persistenceFilePath)
+
+        val descriptorPub = Descriptor(
+            "wpkh([9122d9e0/84'/1'/0']tpubDCYVtmaSaDzTxcgvoP5AHZNbZKZzrvoNH9KARep88vESc6MxRqAp4LmePc2eeGX6XUxBcdhAmkthWTDqygPz2wLAyHWisD299Lkdrj5egY6/0/*)#zpaanzgu",
+            Network.SIGNET
+        )
+
+        val wallet: Wallet = Wallet.load(
+            descriptorPub,
+            changeDescriptor,
```

Comment:
This is one of the quirks of the new API: you can give a private descriptor and a public descriptor, and the extract keys will work (only on the private one) and then you have an odd wallet that can sign but only parts of the transactions if you have UTXOs from both chains, and none if you have UTXOs from your public chain, but also _all_ if you only use UTXOs of the private descriptor chain. Anyway it's not undefined behaviour per se, but it's odd and I'd not recommend it to anyone unless they have a very clear reason for it. I suggest we don't add a test that displays that weird behaviour either.

Let's use 2 public descriptors instead what do you think? Or let me know if you had a reason in mind for using one of each.

Reply:
By the way this is another reason why multipath descriptors are awesome.

Reply:
> Let's use 2 public descriptors instead what do you think? Or let me know if you had a reason in mind for using one of each.

Sounds good, I was just trying it to see what would happen in that case, pushing up changes to the tests to use 2 public descriptors now-